### PR TITLE
Changes to file structure (schema v4.1.0)

### PR DESCRIPTION
**Pull request by calina-c**

### CODE REVIEW

File: ocean_lib/data_provider/data_service_provider.py [hunk 1]
```diff
@@ -519,11 +406,9 @@ def compute_job_result(
...
-        if not response:
-            raise DataProviderException("No response on job result endpoint.")
-
-        if response.status_code != 200:
-            raise DataProviderException(response.content)
+        DataServiceProviderBase.check_response(
+            response, "jobResultEndpoint", compute_job_result_endpoint
+        )
 
         return response.content
```
Comment (mariacarmina) [inline ocean_lib/data_provider/data_service_provider.py:413]: `response.content` returns bytes here, but the signature of this function requires `Dict[str, Any]`.

File: ocean_lib/data_provider/data_service_provider.py [hunk 2]
```diff
@@ -562,185 +447,6 @@ def compute_job_result_logs(
 
         return function_result
```
Comment (mariacarmina) [inline ocean_lib/data_provider/data_service_provider.py:448]: Expected type `Dict[str, Any]` and function_result has `List[Dict[str, Any]]` type.

File: ocean_lib/data_provider/base.py [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+    @staticmethod
+    @enforce_types
+    def check_response(
+        response: Any,
+        endpoint_name: str,
+        endpoint: str,
+        success_codes: Optional[List] = None,
+        exception_type=DataProviderException,
+    ):
+        if not response or not hasattr(response, "status_code"):
```
Comment (mariacarmina) [inline ocean_lib/data_provider/base.py:284]: In `check_response` we can check if the response is None.

Comment (calina-c) [reply ocean_lib/data_provider/base.py:285]: done, I forgot to paste something from the previous PR

File: ocean_lib/data_provider/data_service_provider.py (hunk 2 above)
Comment (calina-c) [reply ocean_lib/data_provider/data_service_provider.py:448]: changed

File: ocean_lib/data_provider/base.py (hunk 3 above)
Comment (calina-c) [reply ocean_lib/data_provider/base.py:285]: changed

**Review by alexcos20 — APPROVED** [review]
lgtm

**Pull request merged**